### ISSUE TITLE
Runner has wrong nesting of describe-it blocks when using IsolationMode.InstancePerLeaf

### ISSUE THREAD

**Issue by StragaSevera** [kotest/kotest#5527]
I have this code:
```kotlin
package ru.ought.advent_of_code_2019

import io.kotlintest.IsolationMode
import io.kotlintest.shouldBe
import io.kotlintest.specs.DescribeSpec

class Day01Test : DescribeSpec() {
    override fun isolationMode(): IsolationMode = IsolationMode.InstancePerLeaf

    private val sut = Day01(2)

    init {
        describe("Day 01") {
            it("has mass") {
                sut.mass.shouldBe(2)
            }

            it("changes mass") {
                sut.mass = 3
                sut.mass.shouldBe(3)
            }

            it("has mass2") {
                sut.mass.shouldBe(2)
            }
        }
    }
}
```

And it results in this output: https://i.imgur.com/eljGuo9.png
All three tests should be grouped under the Describe block, not only first test. This problem exists only if using IsolationMode.InstancePerLeaf mode.

**Comment by netvl** [comment]
I observe the same issue with the `InstancePerTest` mode too. `SingleInstance` works fine.

**Comment by sksamuel** [comment]
This is fixed in kotest 4.0.2
